packages/presets/src/fragments/dual-links/dual-links.ts

```diff
@@ -0,0 +1,452 @@
+import type { ListService } from '@blocksuite/blocks';
+import {
+  type AffineTextAttributes,
+  getAffineInlineSpecsWithReference,
+  ReferenceNodeConfig,
+} from '@blocksuite/blocks';
+import { BlocksUtils, InlineManager, RichText } from '@blocksuite/blocks';
+import { assertExists, noop } from '@blocksuite/global/utils';
+import type { DeltaInsert } from '@blocksuite/inline';
+import { WithDisposable } from '@blocksuite/lit';
+import type { Page } from '@blocksuite/store';
+import { css, html, LitElement, nothing, type TemplateResult } from 'lit';
+import { customElement, property, state } from 'lit/decorators.js';
+import { repeat } from 'lit/directives/repeat.js';
+import { styleMap } from 'lit/directives/style-map.js';
+
+import {
+  ArrowJumpIcon,
+  ArrowLeftIcon,
+  SmallLinkedPageIcon,
+} from '../_common/icons.js';
+
+noop(RichText);
+
+const { matchFlavours } = BlocksUtils;
+
+@customElement('dual-links')
+export class DualLinks extends WithDisposable(LitElement) {
```

Comment:
The term "dual-link" is not widely used in PKMs, we'd refer to it as "bi-directional links". So name the fragment as `BiDirectionalLinkPanel` should be more appropriate.